src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+    /// Delegate Deployer contract directly after it creates the clone for each new project.
+    /// @param _projectId The ID of the project this contract's functionality applies to.
+    function initialize(
+        uint256 _projectId,
+        uint256 _cycleTarget,
+        uint256 _minimumPledgeAmount,
+        IJBController3_1 _controller,
+        IJBSingleTokenPaymentTerminalStore3_1_1 _paymentTerminalStore
+    ) external {
+        // Stop re-initialization.
+        if (projectId != 0) revert ContractAlreadyInitialized();
+
+        // Store project parameters and ready JB architecture contracts for calling
+        cycleTarget = _cycleTarget;
+        minimumPledgeAmount = _minimumPledgeAmount;
+        projectId = _projectId;
+        controller = _controller;
+        paymentTerminalStore = _paymentTerminalStore;
+        directory = controller.directory();
+        fundAccessConstraintsStore = controller.fundAccessConstraintsStore();
+        address terminalAddr = address(directory.terminalsOf(projectId)[0]);
+        paymentTerminal = IJBSingleTokenPaymentTerminal(terminalAddr);
+        //paymentToken = paymentTerminal.token();
+        (JBFundingCycle memory cycleData,) = controller.currentFundingCycleOf(projectId);
+        cycleExpiryDate = cycleData.start + cycleData.duration;
+    }
+
+    /**
+     * @notice Project creator calls this function to deposit refund bonus before funding cycle start
+     * to increase potential pledger confidence.
+     * @dev Can only be called by the contract owner / project creator.
+     */
+    function depositRefundBonus(uint256 refundBonusAmount) external payable onlyOwner {
+        // Reverts if initialize() has not been called yet.
+        if (projectId == 0) revert DataSourceNotInitialized();
+
+        // Reverts if ETH sent does not match input amount.
+        if (msg.value != refundBonusAmount) revert FundsMustMatchInputAmount(refundBonusAmount);
+
+        // Updates contract refund bonus variable.
+        totalRefundBonus = refundBonusAmount;
+
+        emit RefundBonusDeposited(msg.sender, totalRefundBonus);
+    }
+
+    /// @notice This function gets called when the project receives a payment.
+    /// @dev Part of IJBFundingCycleDataSource.
+    /// @dev This implementation just sets this contract up to receive a `didPay` call.
+    /// @param _data The Juicebox standard project payment data. See https://docs.juicebox.money/dev/api/data-structures/jbpayparamsdata/.
+    /// @return weight The weight that project tokens should get minted relative to. This is useful for optionally customizing how many tokens are issued per payment.
+    /// @return memo A memo to be forwarded to the event. Useful for describing any new actions that are being taken.
+    /// @return delegateAllocations Amount to be sent to delegates instead of adding to local balance. Useful for auto-routing funds from a treasury as payment come in.
+    function payParams(JBPayParamsData calldata _data)
```

Comment:
should this check to make sure time is within the cycle?

Reply:
should chain to view since not changing contract data

Reply:
> should this check to make sure time is within the cycle?
I'll need to think on this and talk with the JB devs if it matters or not or causes unintended side effects if `payParams()` is directly called away from the JB platform. The critical functions that need to be time and/or caller gated are the `didPay()` and both redeem functions. `payParams` is a function that gets called from within the `JBSingleTokenPaymentTerminalStore3_1.recordPaymentFor()` call, which will be auto-gated by the project launch parameters. If someone calls `payParams` directly, it should just return the variables to the EVM I guess. But I'll double check with the devs for sure.

Reply:
> should change to view since not changing contract data

Done!

Reply:
since it's a view, i wouldn't worry about reverting. I tend to dislike view functions that revert unless there's a real reason to do so.